### PR TITLE
add typescript overview

### PR DESCRIPTION
**Pull request by jeremyw24**
Needs to be nested under Overview > Languages.

Cross referenced with @MrYuion for accuracy.

### CODE REVIEW

File: overview/typescript.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+## TypeScript References and Tooling
+
+The [TypeScript Handbook](https://www.TypeScriptlang.org/docs/handbook) provides detailed documentation specifically related to TypeScript. 
+
+PlaceOS use [NX](https://nx.dev/) which is a suite of powerful tools to assist with testing and building JavaScript Applications.
+
+NX adds [Jest](https://jestjs.io/) for Unit Testing and [Cypress](https://www.cypress.io/) for Integration and End-to-End Testing.
+
+The [Angular CLI](https://angular.io/cli) is a command-line interface that lets you develop, scaffold and maintain Angular applications and is used with PlaceOS Frontend Applications.
```
Comment (mghill) [inline overview/typescript.md]: Is this referring to https://github.com/PlaceOS/frontends ? If so it should be pluralized in both instances where it's marked as an error

File: overview/typescript.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+---
+title: TypeScript
+description: TypeScript is an open-source language which builds on JavaScript
+sidebar_position: 8
+---
+
+## Overview
+
+TypeScript is an open-source language which builds on JavaScript, one of the world’s most used tools, by adding static type definitions.
```
Comment (mghill) [inline overview/typescript.md]:
```suggestion
TypeScript is an open-source language which builds on JavaScript by adding static type definitions.
```

File: overview/typescript.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+Types provide a way to describe the shape of an object, providing better documentation, and allowing TypeScript to validate that your code is working correctly.
```
Comment (mghill) [inline overview/typescript.md]: Are documentation and validation consequences of describing the shape? This is ambiguous and might be better as dot points.

File: overview/typescript.md (hunk 1 above)
Comment (jeremyw24) [reply overview/typescript.md]: No, not specifically, its just referring to the tool and the fact that we leverage it.

File: overview/typescript.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+## TypeScript and Angular
+
+Angular applications can only be built with TypeScript.
+As Angular is the predominant language used for PlaceOS Frontend Applications this makes TypeScript a requirement.
```
Comment (mghill) [inline overview/typescript.md]: Frontend should be lower case if talking about frontend applications in general, or pluralised if it is talking about 
https://github.com/PlaceOS/frontends

**Review by mghill — CHANGES_REQUESTED** [review]
I'm generally happy to do a copy-edit once over, but had a few questions which aren't straightforward fixes

File: overview/typescript.md (hunk 3 above)
Comment (jeremyw24) [reply overview/typescript.md]: Copied it verbatim from TypeScripts own website, do what you want with it.

File: overview/typescript.md (hunk 1 above)
Comment (mghill) [reply overview/typescript.md]: sorry, submitted a comment separate to review. In the review comment I made it clearer I'm referring to "Fronted Applications" - if it is applications in general then it should be lower case, if it is PlaceOS Frondends, it should be pluralised

File: overview/typescript.md (hunk 4 above)
Comment (jeremyw24) [reply overview/typescript.md]: I have changed this to lower case but not plural, these docs are designed for anyone to write their own front end and may not be using and/or referencing https://github.com/PlaceOS/frontends

Comment (mghill) [reply overview/typescript.md]: perfect

**Review by mghill — APPROVED** [review]

**Pull request merged**